src/sentry/profiles/task.py

```diff
@@ -496,13 +576,14 @@ def truncate_stack_needed(
             - len(symbolicated_frames_dict)
         )
 
-        assert len(new_frames) == new_frames_count
+        if platform in SHOULD_SYMBOLICATE_JS:
+            assert len(new_frames) == new_frames_count
```

Comment:
Is there a reason why this assertion doesn't hold for react native cocoa frames?

Reply:
probably because native has inline frames and thus the number of frames in total can grow.

Reply:
That should be accounted for in this assertion from what I can see.

Reply:
This is more subtle and it took me a little debugging to catch.

The reason while this doesn't hold for react-native is that in case we're dealing with a cocoa stack, we make a copy of the leaf frame with `adjust_instruction_addr = False` 

If the original frame doesn't happen to shows up in the middle of another stack, then it'll never be used.

Reply:
In that case, can we leave a very clear comment explaining why this assertion is not applicable here? Opting something out of sanity checks without explanation is not great

Reply:
Good point, I'll add a comment there.